packages/vite/src/node/plugins/assetImportMetaUrl.ts

```diff
@@ -29,20 +24,19 @@ export function assetImportMetaUrlPlugin(config: ResolvedConfig): Plugin {
         code.includes('new URL') &&
         code.includes(`import.meta.url`)
       ) {
-        const importMetaUrlRE =
-          /\bnew\s+URL\s*\(\s*('[^']+'|"[^"]+"|`[^`]+`)\s*,\s*import\.meta\.url\s*,?\s*\)/g
-        const noCommentsCode = code
-          .replace(multilineCommentsRE, blankReplacer)
-          .replace(singlelineCommentsRE, blankReplacer)
-          .replace(stringsRE, (m) => `'${'\0'.repeat(m.length - 2)}'`)
-
-        let s: MagicString | null = null
+        let s: MagicString | undefined
         let match: RegExpExecArray | null
-        while ((match = importMetaUrlRE.exec(noCommentsCode))) {
+        const assetImportMetaUrlRE =
+          /\bnew\s+URL\s*\(\s*('[^']+'|"[^"]+"|`[^`]+`)\s*,\s*import\.meta\.url\s*,?\s*\)/g
+        const cleanString = emptyString(code)
+        while ((match = assetImportMetaUrlRE.exec(cleanString))) {
```

Comment:
I think I would prefer the `match` variable to be closer to the while loop like before 🤔 I find it easier to relate as `match` is only temporarily used to iterate over the regex.